bson/types.go

```diff
@@ -7,44 +7,111 @@
 package bson
 
 import (
-	"go.mongodb.org/mongo-driver/bson/bsontype"
+	"encoding/json"
+	"net/url"
+	"reflect"
+	"time"
+
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
 )
 
+// Type represents a BSON type.
+type Type bsoncore.Type
```

Comment:
We should not export a type that's canonically defined in an "x" package. This should be `type Type byte`.